### ISSUE TITLE
How users can continue to use Core type addon(s) ?

### ISSUE THREAD

**Issue by MlemandPurrs** [wesnoth/wesnoth#5619]
There used be a type of UMC addon possible, the Core type addons. One of such type i had made during 1.14 versions, the 'Alternate Default Core', however recently it came to my attention the button for switching Cores had been removed from the main menu. Apparently there is only a hotkey left. Now im inclined to ask how can players continue to use this type of addon ?

**Comment by knyghtmare** [comment]
Screenshot showing 2 cores in add-ons directory:
![Screenshot 2021-03-15 184809](https://user-images.githubusercontent.com/5283677/111158272-09d47400-85c2-11eb-9d8f-aa86378151f5.png)
Screenshot showing absence of any option to switch cores:
![Screenshot 2021-03-15 184827](https://user-images.githubusercontent.com/5283677/111158279-0b05a100-85c2-11eb-9f32-1544803338b8.png)

I was the one who pointed this out so I am just posting the confusion I got.

**Comment by CelticMinstrel** [comment]
As you mentioned, currently the only way to manage cores is to assign a hotkey, like this:

![image](https://user-images.githubusercontent.com/578731/111269371-e979d900-8604-11eb-933b-d8726a734c9a.png)

Add a hotkey (whatever you want), then hit close and press that hotkey. You should be presented with a list of available cores.

It's perhaps a little overly obscure this way, but the functionality _is_ still there. I'm not sure what would be a better place to put it without cluttering up the title screen as it used to. I suppose it could be added somewhere in preferences or something.

**Comment by shikadiqueen** [comment]
I swear we decided to make the button show up somewhere when alternative cores were present. @Vultraz?

**Comment by CelticMinstrel** [comment]
Yeah, at one point the Cores button would appear in the main menu if you had an alternate core installed, but that doesn't seem to be the case now - I checked the code and there's no mention of a Cores button.

**Comment by Vultraz** [comment]
Pretty sure that was removed at some point.

**Comment by CelticMinstrel** [comment]
Sure, but _why_ was it removed?

**Comment by Pentarctagon** [comment]
It looks like it was removed in 22acd3615591565bd5960a471a0ca6f71e24e69e, 1fae8c238da619a61cd4778b1af0201a174e4998, and 5302711439872ebc944656dfaae9352408c33103.  So re-adding it would be trivial I assume, though I also wonder if it wouldn't be better to deprecate and eventually remove the Core type add-on entirely at some point as well.  It's a rarely used type of add-on, rarely if ever a good solution to whatever someone wants to do, and seems primarily aimed at a use case that few except the now-vanished author of the feature thought would be useful to have.

**Comment by soliton-** [comment]
> rarely if ever a good solution to whatever someone wants to do

I don't know what that refers to. To me Cores are there to support total conversions for which there is no other solution at all currently.

I don't mind that not every other addon author wants to make a total conversion would be weird if that were so but I don't see that as much of an argument against the feature. Especially since at least from what I've seen the feature is really pretty simple implementation wise.

To me the basic idea is to have a way to let an addon author decide what config files wesnoth loads at startup, including those that are otherwise hardcoded. I'm curious if people really don't like that idea or the current implementation or what? (The current implementation no doubt lacks polish and can be improved but it implements the basic idea just fine.)

**Comment by Pentarctagon** [comment]
Well, that's the thing - cores have existed for 7+ years, and nobody has actually made any total conversion add-ons that I'm aware of.  Certainly there's none on the current 1.14 or 1.15 add-on servers.  So the feature itself isn't being used for its intended purpose anyway.

Of the currently existing cores:
* Alternate AI - changes the default AI to the experimental AI, which afaik had most of its improvements added to the default AI in 1.15 anyway and so isn't needed anymore.
* Alternate Default Core - the first sentence of the description says "This addon is rather intended just as an example of an working core type addon.", and nothing it lists as features sounds like it can't already be done with eras and modifications since it's combining things from other eras and modifications.
* Slim Core - it's the default core, but disables all mainline units, races, traits, abilities, and weapon specials.  Its reason for existing seems to be that "This allows to use your own units in a fresh environment.", but if you don't want to use any of those things from mainline then you can stay with the default core, ignore mainline units, abilities, and weapon specials, and if needed create a separate race to avoid the default traits.  So again, this could already be accomplished without needing a Core.

More fundamentally though, I simply don't think that making total conversions (aka essentially making a brand new game on top of Wesnoth's engine) is a feature that should be supported.  We certainly don't have enough active developers to support that kind of use if it ever did actually happen.

**Comment by soliton-** [comment]
What would you call [Galactic  Empires](https://forums.wesnoth.org/viewtopic.php?p=309052#p309052) for example?


> We certainly don't have enough active developers to support that kind of use if it ever did actually happen.

I don't understand what we would have to do?

**Comment by Pentarctagon** [comment]
> What would you call Galactic Empires for example?

Well, Galactic Empires calls itself an era and contains some maps/scenarios, so I'd call it an MP era/map pack.  I'd also say the fact that it says in its description that it's "more or less a total conversion mod" without needing to be a Core shows that the Core add-on type is generally not needed.

> I don't understand what we would have to do?

I think it's a safe assumption to make that if someone tries to make full use of a feature that has never been fully used before, remains relatively undocumented, and has been effectively unmaintained for several years, that there will be a variety of bugs found and additional features requested.
